pkg/services/searchV2/filter.go

```diff
@@ -74,28 +75,35 @@ func (q *PermissionFilter) canAccess(kind entityKind, id string) bool {
 	//
 	switch kind {
 	case entityKindFolder:
-		if id == "" {
+		if id == folder.GeneralFolderUID {
```

Comment:
hmm.. I see two constants: one for `GeneralFolderUID`, the other for the [`RootFolderUID`](https://github.com/grafana/grafana/blob/b573b19ca379a2b38f7edf99cd1f421ea74f29c2/pkg/services/folder/model.go#L19) which matches the previous magic empty string. I am not sure which one is correct here

Reply:
I think the general folder is the correct one here, both from testing and from looking at the code (see the comments above with links to where `GeneralFolderUID` is set).

Reply:
I don't think "GeneralFolderUID" is a real thing 🤔  -- all existing code uses "" to mean the root folder, and it is called "General" in the UI

Reply:
Unfortunately it is and it seems it might be an issue we have to resolve, there are parts of the code using it.

RBAC defined the scope for General/Root Folder access as `folders:uid:general` and translated "" folder UID to the "general" scope (correct me if I'm wrong) since it made it awkward to have no human readable identifier for the general folder.

 but I now see this being used in the folder service for general folder UID checks ([example](https://github.com/grafana/grafana/blob/cef15873d6f56bcc7264280db41afaf89100bbfb/pkg/services/folder/folderimpl/sqlstore.go#L54)) 
 
 From the example I sent, if dashboards are being store with root/general folder as "" then the folder exists check is not happening
 
 Pinging @zserge on this, we might need to settle this and check the sanity of the dashboard table in a migration.